packages/dds/sequence/src/revertibles.ts

```diff
@@ -0,0 +1,168 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { unreachableCase } from "@fluidframework/common-utils";
+import { LocalReferencePosition, PropertySet } from "@fluidframework/merge-tree";
+import { IntervalCollection, SequenceInterval } from "./intervalCollection";
+import { SharedString } from "./sharedString";
+
+const IntervalEventType = {
+	CHANGE: 0,
+	ADD: 1,
+	DELETE: 2,
+	PROPERTYCHANGED: 3,
+} as const;
+
+type IntervalEventType = typeof IntervalEventType[keyof typeof IntervalEventType];
+
+type IntervalRevertible =
+	| {
+			event: typeof IntervalEventType.CHANGE;
+			interval: SequenceInterval;
+			start: LocalReferencePosition;
+			end: LocalReferencePosition;
+			collection: IntervalCollection<SequenceInterval>,
+	  }
+	| {
+			event: typeof IntervalEventType.ADD;
+			interval: SequenceInterval;
+			collection: IntervalCollection<SequenceInterval>,
+	  }
+	| {
+			event: typeof IntervalEventType.DELETE;
+			interval: SequenceInterval;
+			start: LocalReferencePosition;
+			end: LocalReferencePosition;
+			collection: IntervalCollection<SequenceInterval>,
+	  }
+
+	| {
+			event: typeof IntervalEventType.PROPERTYCHANGED;
+			interval: SequenceInterval;
+			propertyDeltas: PropertySet;
+	  };
+
+type TypedRevertible<T extends IntervalRevertible["event"]> =
+	IntervalRevertible & {event: T;};
+
+function appendLocalAddToRevertibles(
```

Comment:
Since there are separate events for each interval action (unlike SharedString using sequenceDelta events for all actions), is that why we don't need a function analogous to appendToMergeTreeDeltaRevertibles?

Reply:
Yes, and they correspondingly take different parameters.